### PR TITLE
Refactor: HTML uniformization, UX small improvement, bug fixes and tests refactor

### PR DESCRIPTION
**Pull request by stafyniaksacha**
Changes: 
* Remove index selector on non related with index pages
* Index selector helper (collection & data menu entry are always visible)
* Sidebar on watch data page are now toggleable (persisted in user session)
* Widget on metrics page are now toggleable (non persisted)
* Pagination added on all listing pages

Refactor:
* Uniformization of structure of all pages
* Uniformization of documents listing (security, collections, data, indexes)
* Uniformization of action icons & titles of actions
* Rewrite all tests about navigation (I go to ..., I am on ....)

Fixes:
* Can't edit JSON Edit fields on advanced search page
* Metadata are now displayed in the watch data page
* Clear filters on basic & advanced filters area
* Realtime/persisted collections distinctions on watch data page
* Small other things ...

### CODE REVIEW

File: core/views/collection/index.hbs [hunk 1]
```diff
@@ -1 +1,31 @@
-<div ui-view="mainView" class="view-fade"></div>
\ No newline at end of file
+<div ui-view="mainView" class="view-fade">
+  <div class="collection-selectIndex">
+    <headline title="Collections" subtitle=""></headline>
+    <img src="https://s-media-cache-ak0.pinimg.com/236x/dd/98/a9/dd98a9a89c10a88b9e3dd52cce6f12b2.jpg" class="" style="
```
Comment (xbill82) [inline core/views/collection/index.hbs]: We should pull this asset to our repo, if we want to integrate it to the project.

File: core/views/indexes/browse.hbs [hunk 2]
```diff
@@ -34,24 +34,45 @@
...
+                                  <cog-options-indexes
+                                      index="index"
+                                      can-edit="false"
+                                      can-delete="canDeleteIndex(index)"
+                                      after-delete="afterDelete(index)">
+                                  </cog-options-indexes>
+                                </div>
                               </div>
+                            </div>
+                            <!--<div class="panel-heading">
```
Comment (xbill82) [inline core/views/indexes/browse.hbs]: Wipe this comment here?

File: core/views/storage/index.hbs [hunk 3]
```diff
@@ -1 +1,33 @@
-<div ui-view="mainView" class="view-fade"></div>
\ No newline at end of file
+<div ui-view="mainView" class="view-fade">
+
+  <div class="storage-browse">
+    <headline title="Storage" subtitle=""></headline>
+    <img src="https://s-media-cache-ak0.pinimg.com/236x/dd/98/a9/dd98a9a89c10a88b9e3dd52cce6f12b2.jpg" class="" style="
```
Comment (xbill82) [inline core/views/storage/index.hbs]: Internalize here

File: features/authentication.feature [hunk 4]
```diff
@@ -4,19 +4,17 @@ Feature: Test authentication
 
   Scenario: Log a user in when she provides correct credentials
     Given I go to the login page
-    Then I see the login page
-    And I authenticate as "admin"
+    Then I authenticate as "admin"
```
Comment (xbill82) [inline features/authentication.feature]: This should be `And`

File: features/authentication.feature [hunk 5]
```diff
@@ -4,19 +4,17 @@ Feature: Test authentication
...
     Then I am authenticated
 
   Scenario: The user is logged out when she presses the logout button
     Given I go to the login page
-    And I authenticate as "admin"
+    Then I authenticate as "admin"
```
Comment (xbill82) [inline features/authentication.feature]: This should be `And`

File: features/browseData.feature [hunk 6]
```diff
@@ -10,17 +10,17 @@ Feature: Test browse data pages
...
-    Given I am on browse data page
+    Given I go to the browse data page
     Then I click on the collection selector
     Then I click on the test collection
     Then I have a list with "2" elements
 
-  Scenario: I will be redirected to indexes browse page if i try to select an wrong collection
-    Given I am on browse data page with an wrong index
-    Then I am on manage index page
+ # Scenario: I will be redirected to indexes browse page if i try to select an wrong collection
```
Comment (xbill82) [inline features/browseData.feature]: Wipe comment here?

File: features/browseData.feature [hunk 7]
```diff
@@ -10,17 +10,17 @@ Feature: Test browse data pages
 
   @cleanDb
   Scenario: Display data list when a collection is selected
-    Given I am on browse data page
+    Given I go to the browse data page
     Then I click on the collection selector
```
Comment (xbill82) [inline features/browseData.feature]: This should be `And`

File: features/browseData.feature [hunk 8]
```diff
@@ -10,17 +10,17 @@ Feature: Test browse data pages
...
     Then I click on the test collection
```
Comment (xbill82) [inline features/browseData.feature]: This should be `And`

File: features/indexes.feature [hunk 9]
```diff
@@ -8,35 +8,35 @@ Feature: Test multi index features
...
 
-  Scenario: I can select an index with the index selector in sidebar
-    Given I go to manage index page
-    Then No index is selected
-    Then I can not see Storage & Realtime menu entries
-    Then I click on the index selector
-    Then I select an index
-    Then The index "kuzzle-bo-testindex" is selected
-    Then I can see Storage & Realtime menu entries
+ # Scenario: I can select an index with the index selector in sidebar
```
Comment (xbill82) [inline features/indexes.feature]: Comment here?

File: features/indexes.feature [hunk 10]
```diff
@@ -8,35 +8,35 @@ Feature: Test multi index features
...
+ #   Then I click on the index selector
+ #   Then I select an index
+ #   Then The index "kuzzle-bo-testindex" is selected
+ #   Then I can see Storage & Realtime menu entries
 
   Scenario: I can select an index directly in the manage index page
     Given I go to manage index page
-    Then No index is selected
-    Then I can not see Storage & Realtime menu entries
+ #   Then No index is selected
```
Comment (xbill82) [inline features/indexes.feature]: Clean up here?

**Comment by xbill82** [comment]
:+1:

File: core/views/realtime/index.hbs [hunk 11]
```diff
@@ -1,7 +1,7 @@
-<div  class="view-fade h100 posrel">
+<div class="  view-fade h100 posrel">
```
Comment (dbengsch) [inline core/views/realtime/index.hbs]: spaces

**Comment by j33f** [comment]
:+100:

File: public/javascripts/role/browse.ctrl.js [hunk 12]
```diff
@@ -28,7 +28,9 @@ angular.module('kuzzle.role')
       $scope.loadRoles = function () {
         roleApi.list(($scope.currentPage - 1) * $scope.limit, $scope.limit)
           .then(function (response) {
-            $scope.roles = response;
+            console.log(response);
```
Comment (dbengsch) [inline public/javascripts/role/browse.ctrl.js]: console.log

**Comment by dbengsch** [comment]
comment above and +1

**Pull request merged**